src/main/java/org/zalando/nakadi/domain/BatchFactory.java

```diff
@@ -1,16 +1,62 @@
 package org.zalando.nakadi.domain;
 
-import org.json.JSONArray;
+import org.json.JSONException;
 
 import java.util.ArrayList;
 import java.util.List;
 
 public class BatchFactory {
 
-    public static List<BatchItem> from(final JSONArray events) {
-        final List<BatchItem> batch = new ArrayList<>(events.length());
-        for (int i = 0; i < events.length(); i++) {
-            batch.add(new BatchItem(events.getJSONObject(i)));
+    public static List<BatchItem> from(final String events) {
+        final List<BatchItem> batch = new ArrayList<>();
+        StringBuilder sb = new StringBuilder();
```

Comment:
Maybe if it wasn't a string but an array of bytes, then we could just create strings using `String(bytes[] bytes, int offset, int length)`. I'm not sure, I think we should investigate more and reduce memory footprint.